pallets/msa/src/lib.rs

```diff
@@ -533,23 +538,31 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::add_key_to_msa())]
 		pub fn add_key_to_msa(
 			origin: OriginFor<T>,
-			key: T::AccountId,
-			proof: MultiSignature,
+			msa_owner_key: T::AccountId,
+			msa_owner_proof: MultiSignature,
+			new_key: T::AccountId,
+			new_proof: MultiSignature,
 			add_key_payload: AddKeyData,
 		) -> DispatchResult {
-			let who = ensure_signed(origin)?;
+			let _ = ensure_signed(origin)?;
 
-			Self::verify_signature(proof.clone(), key.clone(), add_key_payload.encode())
-				.map_err(|_| Error::<T>::AddKeySignatureVerificationFailed)?;
+			Self::verify_signature(
+				vec![
+					(msa_owner_proof.clone(), msa_owner_key.clone()),
+					(new_proof, new_key.clone()),
```

Comment:
what new_proof?

Reply:
Renamed , also reverted to verify_signature and call it twice in add_key_to_msa